### ISSUE TITLE
Open Source Code Must Be Updated at least to = VS 2017

### ISSUE THREAD

**Issue by steve007iii** [x64dbg/x64dbg#3009]
The source code must be updated to be compiled with more current tools at least to Visual Studio 2017
and QT

The tools are very old
2015/2017/2019/2022  = they are backward compatible 2013 is not

------------------------

**Comment by mrexodia** [comment]
The code builds fine with 2022.

**Comment by requizm** [comment]
@mrexodia Hi, I have a problem with building `x64dbg_dbg` on Visual Studio 2022. Can I ask from here? Or should I create a new issue?

**Comment by mrexodia** [comment]
Sure, what’s the issue?

**Comment by requizm** [comment]
@mrexodia 
I checked steps on doc: https://github.com/x64dbg/x64dbg/wiki/Compiling-the-whole-project

OS: Windows 10
IDE: Visual Studio 2022, msvc 14.34.31933, MFC installed

I have QT 5.6.3 for both architectures in `C:\` and Windows SDK.

When I build with x64 or x86 in VS 2022, I get lots of errors.
```
Severity	Code	Description	Project	File	Line	Suppression State
Error (active)	E0040	expected an identifier	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\iosfwd	24	
Error (active)	E0864	basic_istream is not a template	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\string	26	
Error (active)	E0065	expected a ';'	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring	33	
Error (active)	E0020	identifier "streampos" is undefined	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring	38	
Error (active)	E0020	identifier "streamoff" is undefined	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring	39	
Error (active)	E0020	identifier "streampos" is undefined	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring	229	
Error (active)	E0020	identifier "streamoff" is undefined	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring	230	
Error (active)	E0020	identifier "streampos" is undefined	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring	364	
Error (active)	E0020	identifier "streamoff" is undefined	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring	365	
Error (active)	E0864	basic_ostream is not a template	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring	505	
Error (active)	E0864	_Char_traits_eq is not a template	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring	574	
Error (active)	E0864	basic_stringbuf is not a template	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring	2447	
Error (active)	E0864	basic_istream is not a template	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring	5311	
Error (active)	E0864	basic_istream is not a template	x64dbg_dbg	C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.34.31933\include\xstring	5312	
Error (active)	E0333	defining an implicitly declared member function is not allowed	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	37	
Error (active)	E0135	class "Lexer" has no member "ReadInputFile"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	42	
Error (active)	E0135	class "Lexer" has no member "SetInputData"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	48	
Error (active)	E0135	class "Lexer" has no member "DoLexing"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	55	
Error (active)	E0020	identifier "TokenState" is undefined	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	55	
Error (active)	E0135	class "Lexer" has no member "Test"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	73	
Error (active)	E0135	class "Lexer" has no member "getToken"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	105	
Error (active)	E0135	class "Lexer" has no member "reportError"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	373	
Error (active)	E0135	class "Lexer" has no member "nextChar"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	379	
Error (active)	E0135	class "Lexer" has no member "reportWarning"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	384	
Error (active)	E0135	class "Lexer" has no member "resetLexerState"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	389	
Error (active)	E0135	class "Lexer" has no member "setupTokenMaps"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	407	
Error (active)	E0135	class "Lexer" has no member "TokString"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	437	
Error (active)	E0020	identifier "TokenState" is undefined	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	437	
Error (active)	E0135	class "Lexer" has no member "TokString"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	467	
Error (active)	E0020	identifier "Token" is undefined	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	467	
Error (active)	E0135	class "Lexer" has no member "peekChar"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	497	
Error (active)	E0135	class "Lexer" has no member "readChar"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	510	
Error (active)	E0135	class "Lexer" has no member "checkString"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	524	
Error (active)	E0135	class "Lexer" has no member "signalNewLine"	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	538	
Error (active)	E0040	expected an identifier	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.h	56	
Error (active)	E0020	identifier "Token" is undefined	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.h	61	
Error (active)	E0018	expected a ')'	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.h	61	
Error (active)	E0147	declaration is incompatible with "std::string std::TokString" (declared at line 61)	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.h	62	
Error (active)	E0169	expected a declaration	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.h	64	
Error (active)	E0757	"Token" is not a type name	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.h	73	
Error (active)	E0757	"Token" is not a type name	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.h	74	
Error (active)	E0757	"Token" is not a type name	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.h	75	
Error (active)	E0757	"Token" is not a type name	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.h	76	
Error (active)	E0757	"Token" is not a type name	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.h	77	
Error (active)	E0757	"Token" is not a type name	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.h	81	
Error (active)	E0757	"Token" is not a type name	x64dbg_dbg	C:\Users\requizm\Documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.h	88	
```

I feel like I'm missing some important thing but I couldn't solve.

**Comment by mrexodia** [comment]
You cannot use Qt 5.6.3 with VS2022. Use a more recent Qt 5 version (6 is not supported) with the msvc2017 ABI.

**Comment by requizm** [comment]
> You cannot use Qt 5.6.3 with VS2022. Use a more recent Qt 5 version (6 is not supported) with the msvc2017 ABI.

@mrexodia Thank you for the info. I installed `MSVC 2017(v141, 14.16)` in my visual studio. I'm also installing QT 5.15(LTS)

But when I compile the project with v141, I'm still getting the same errors. Are you sure that errors are related to QT? Because logs are related to _lexer.h, xstring.h, iosfwd, string_ files. If there was an issue with the QT version, wouldn't it be the error? (I'm sure my QT version is wrong but I think I got another error before I even saw QT error)

**Comment by mrexodia** [comment]
Yeah it looks like there was a missing `#include <string>` in `lexer.h`. I'll also update to the latest Visual Studio and see if there are any other issues. In previous versions this was working fine, but they are updating the standard library headers to be less dependent on each other so these type of errors happen from time to time...

**Comment by mrexodia** [comment]
Should be fixed now.

**Comment by requizm** [comment]
@mrexodia 

Thanks, the missing header error is gone. Now we have other errors related to `cctype.h` and `lexer.cpp`

(48 line)
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2059	syntax error: 'extern ' (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\program files (x86)\windows kits\10\include\10.0.20348.0\ucrt\ctype.h	19	
Error	C3805	'type': unexpected token, expected either '}' or a ',' (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\program files (x86)\windows kits\10\include\10.0.20348.0\ucrt\ctype.h	19	
Error	C2059	syntax error: 'extern ' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files (x86)\windows kits\10\include\10.0.20348.0\ucrt\ctype.h	19	
Error	C3805	'type': unexpected token, expected either '}' or a ',' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files (x86)\windows kits\10\include\10.0.20348.0\ucrt\ctype.h	19	
Error	C2061	syntax error: identifier 'unique_ptr' (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\ast.h	11	
Error	C2649	'typename': is not a 'class' (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\ast.h	13	
Error	C2631	'StatDecl': a class or enum cannot be defined in an alias template (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\ast.h	33	
Error	C2988	unrecognizable template declaration/definition (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\ast.h	94	
Error	C2059	syntax error: '<end Parse>' (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\ast.h	94	
Error	C2631	'Parser': a class or enum cannot be defined in an alias template (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\parser.h	7	
Error	C1004	unexpected end-of-file found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\parser.cpp	189	
Error	C2039	'isalnum': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	34	
Error	C2873	'isalnum': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	34	
Error	C2039	'isalpha': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	34	
Error	C2873	'isalpha': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	34	
Error	C2039	'iscntrl': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	34	
Error	C2873	'iscntrl': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	34	
Error	C2039	'isdigit': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	35	
Error	C2873	'isdigit': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	35	
Error	C2039	'isgraph': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	35	
Error	C2873	'isgraph': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	35	
Error	C2039	'islower': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	35	
Error	C2873	'islower': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	35	
Error	C2039	'isprint': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	36	
Error	C2873	'isprint': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	36	
Error	C2039	'ispunct': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	36	
Error	C2873	'ispunct': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	36	
Error	C2039	'isspace': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	36	
Error	C2873	'isspace': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	36	
Error	C2039	'isupper': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	37	
Error	C2873	'isupper': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	37	
Error	C2039	'isxdigit': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	37	
Error	C2873	'isxdigit': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	37	
Error	C2039	'tolower': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	37	
Error	C2873	'tolower': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	37	
Error	C2039	'toupper': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	38	
Error	C2873	'toupper': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	38	
Error	C2039	'isblank': is not a member of '`global namespace'' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	40	
Error	C2873	'isblank': symbol cannot be used in a using-declaration (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files\microsoft visual studio\2022\community\vc\tools\msvc\14.16.27023\include\cctype	40	
Error	C3861	'isprint': identifier not found (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\helpers.h	54	
Error	C3861	'isspace': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	108	
Error	C3861	'isxdigit': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	170	
Error	C3861	'isxdigit': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	236	
Error	C3861	'isalpha': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	258	
Error	C3861	'isalnum': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	262	
Error	C3861	'isxdigit': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	282	
Error	C3861	'isdigit': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	294	
Error	C3861	'isdigit': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	298	
```


I also replaced `cctype` with `ctype.h`. But there are still similar errors:
(20 line)
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	C2059	syntax error: 'extern ' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files (x86)\windows kits\10\include\10.0.20348.0\ucrt\ctype.h	19	
Error	C2059	syntax error: 'extern ' (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\program files (x86)\windows kits\10\include\10.0.20348.0\ucrt\ctype.h	19	
Error	C3805	'type': unexpected token, expected either '}' or a ',' (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\program files (x86)\windows kits\10\include\10.0.20348.0\ucrt\ctype.h	19	
Error	C3805	'type': unexpected token, expected either '}' or a ',' (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\program files (x86)\windows kits\10\include\10.0.20348.0\ucrt\ctype.h	19	
Error	C2061	syntax error: identifier 'unique_ptr' (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\ast.h	11	
Error	C2649	'typename': is not a 'class' (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\ast.h	13	
Error	C2631	'StatDecl': a class or enum cannot be defined in an alias template (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\ast.h	33	
Error	C2988	unrecognizable template declaration/definition (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\ast.h	94	
Error	C2059	syntax error: '<end Parse>' (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\ast.h	94	
Error	C2631	'Parser': a class or enum cannot be defined in an alias template (compiling source file btparser\btparser\parser.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\parser.h	7	
Error	C1004	unexpected end-of-file found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\parser.cpp	189	
Error	C3861	'isprint': identifier not found (compiling source file btparser\btparser\lexer.cpp)	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\helpers.h	54	
Error	C3861	'isspace': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	108	
Error	C3861	'isxdigit': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	170	
Error	C3861	'isxdigit': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	236	
Error	C3861	'isalpha': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	258	
Error	C3861	'isalnum': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	262	
Error	C3861	'isxdigit': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	282	
Error	C3861	'isdigit': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	294	
Error	C3861	'isdigit': identifier not found	x64dbg_dbg	c:\users\requizm\documents\projects\github\x64dbg\x64dbg\src\dbg\btparser\btparser\lexer.cpp	298	
```

**Comment by mrexodia** [comment]
Feel free to fix them, I don’t have them over here with the latest VS2022…

**Comment by mrexodia** [comment]
I also tested with `/std:latest /permissive-` and for good measure also when building with LLVM.

**Comment by requizm** [comment]
Alright, it's fixed. Yesterday when I pulled after the last commit in btparser, there was a problem in the submodule. When I fixed it, other problems went away. Thank you for support.

Also good news, I'm able to compile with v143(MSVC2022)

Note to the person reading this:  
**Visual Studio 2022 (MSVC 2022) compile guide:**
1. Install VS2022 C++ with MFC and Windows 10 SDK.
1. Install Qt 5 recent version. In my case, `5.15.2`. We can install it with Qt Online Installer. Select the last MSVC 32 and 64 bit components in the setup wizard. In my case, MSVC2019. Also, select Qt Creator.
1. Compile solution with VS2022. 
1. Open the `src\gui\x64dbg.pro`with Qt Creator. We may need to manually set the compiler as written in the documentation, but it was found automatically in my case. Then build.
1. If we try to run with Qt Creator, it should say there is no executable file. So we need to set Executable File. 
1. If we try to run again, it should give DLL errors. Copy all DLLs from `path/to/QT/msvc29/bin` to the folder that contains the executable file. Then copy `path/to/QT/msvc29/plugins/imageformats` and `path/to/QT/msvc29/plugins/platforms` folders to the folder that contains the executable file.
